### PR TITLE
New PF iso recomputation from miniAOD histos 81X V2

### PR DESCRIPTION
**Pull request by archiron**
same as PR14855

Adding histograms to check the reproductibility of the PF isolation recomputation from miniAOD.

These are included in file ElectronMcMiniAODSignalValidator.cc (h) and in file ElectronMcSignalValidatorMiniAOD_cfi.py and in a new file ElectronIsolation_cfi.py which is taken into account in electronValidationSequenceMiniAOD_cff.py

Other minor modifications or corrections are made in the others files (Validation/RecoEgamma/test) to create these histos into the web page.

### CODE REVIEW

**Comment by dmitrijus** [comment]
+1

**Comment by archiron** [comment]
seems to be a DAS pbm. I got the same result this morning, but in june the 13 the test pass :
element = 140.53
processing relval_standard
ignoring non-requested file relval_highstats
ignoring non-requested file relval_pileup
ignoring non-requested file relval_generator
ignoring non-requested file relval_extendedgen
ignoring non-requested file relval_production
ignoring non-requested file relval_ged
ignoring non-requested file relval_upgrade
ignoring non-requested file relval_2017
ignoring non-requested file relval_identity
ignoring non-requested file relval_machine
ignoring non-requested file relval_unsch
ignoring non-requested file relval_premix
Running in 4 thread(s)

Preparing to run 140.53 RunHI2011+RunHI2011+RECOHID11+HARVESTDHI
cleaning up  140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI  in  /afs/cern.ch/work/a/archiron/private/CMSSW_8_1_0_pre5_ValELE/src

# in: /afs/cern.ch/work/a/archiron/private/CMSSW_8_1_0_pre5_ValELE/src going to execute cd 140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI
 echo '{
"182124":[[1,268435455]]
}' > step1_lumiRanges.log  2>&1
 
# in: /afs/cern.ch/work/a/archiron/private/CMSSW_8_1_0_pre5_ValELE/src going to execute cd 140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI
 (das_client.py --limit 0 --query 'file dataset=/HIMinBiasUPC/HIRun2011-v1/RAW run=182124 site=T2_CH_CERN') | sort -u > step1_dasquery.log  2>&1
---
# in: /afs/cern.ch/work/a/archiron/private/CMSSW_8_1_0_pre5_ValELE/src going to execute cd 140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI
 cmsDriver.py step2  --conditions auto:run1_data -s RAW2DIGI,L1Reco,RECO,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBiasHI+TkAlUpsilonMuMuHI+TkAlZMuMuHI+TkAlMuonIsolatedHI+TkAlJpsiMuMuHI+HcalCalMinBias,DQM --process reRECO -n 30 --data  --eventcontent RECO,DQM --runUnscheduled  --scenario HeavyIons --datatier RECO,DQMIO --repacked  --filein filelist:step1_dasquery.log --lumiToProcess step1_lumiRanges.log --fileout file:step2.root  > step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log  2>&1
 
# in: /afs/cern.ch/work/a/archiron/private/CMSSW_8_1_0_pre5_ValELE/src going to execute cd 140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI
 cmsDriver.py step3  --data  --filetype DQM --conditions auto:run1_data --scenario HeavyIons -s HARVESTING:dqmHarvesting -n 100  --filein file:step2_inDQM.root --fileout file:step3.root  > step3_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log  2>&1
 
140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI Step0-PASSED Step1-PASSED Step2-PASSED  - time date Mon Jun 13 11:53:02 2016-date Mon Jun 13 11:45:54 2016; exit: 0 0 0
1 1 1 tests passed, 0 0 0 failed

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h [hunk 1]
```diff
@@ -52,6 +77,13 @@ class ElectronMcSignalValidatorMiniAOD : public ElectronDqmAnalyzerBase {
       std::vector<int> matchingMotherIDs_;
       std::string outputInternalPath_ ;
 
+      Float_t sumChargedHadronPt_recomp;
```
Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: @archiron - please use native floats

Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: OK. Done.

**Comment by dmitrijus** [comment]
+1

**Comment by archiron** [comment]
seems to be "DAS related" as I see on others PR

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h [hunk 2]
```diff
@@ -17,6 +17,8 @@
 
 #include "DataFormats/VertexReco/interface/VertexFwd.h"
 #include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
```
Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: @archiron - what creates the need for all of these new includes?

Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: there is no need of it. It was include when I added the recomp modifications from another file by copy-paste. Do you want I remove it and resub ?

Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: yes - please include only the headers you need for your code.

Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: Done. But I always have the "Failed tests: RelVals" error with the 140.53

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**